### ISSUE TITLE
Multi-dataset learning

### ISSUE THREAD

**Issue by jacobmiller29** [ThilinaRajapakse/simpletransformers#1081]
I have a fairly complicated use case. To summarize, I have a (very) large database of text, call it A, that I want to run inference on. However, I do not have labels for this database. I instead use a much smaller but closely related database that does have labels, call it B, to fine-tune a model. I then run this fine-tuned model on the large database to get predictions. 

I do this in 3 steps: 
(1) Firstly, to reduce noise, I fine-tune the model on A with a language modeling task. (I've found this first step makes a big difference in my use case)
(2) I then fine-tune this model on a classification task on B.
(3) Lastly, I then run the classification model on A to get the predictions.

The problem I keep running into is that at step (2), the model is expecting the training data to be the same size as in step (1), which is not the case. I bypass this problem at the moment by taking "random samples" of data from A equal in size to the training set in B and then training many times, but this is obviously suboptimal. 

I'm wondering if I'm missing a simple way to reconfigure the model after step (1) so that it expects the training data in step (2) to be the correct size? Having gone closely through the documentation, I haven't been able to figure out a way to do this.

**Comment by AbinayaM02** [comment]
@jacobmiller29 : 
1. I'm assuming "training data to be same size" as "training data should have same shape".  On A, the no. of output label is different from the no. of output label on the dataset B. Is this understanding correct? If so, in the first step, what's the output shape of the model?
2. What's the output shape of the second model (dataset B)?

**Comment by jacobmiller29** [comment]
Thanks for your reply! Yes, you're correct, the expected number of output labels for dataset A is ~6 million, while for dataset B it is ~6000. 

The problem is that in step 2 when I try to fine-tune on dataset B with a classification task, the model always skips training unless I set the number of epochs to be very high. 

This must be because the model configures an epoch in step 1 based on the length of training data in A. When I fine-tune on a new task in step 2, the model is expecting the training data to be much longer. It seems to me that when a model has a new head added to it, this "parameter" should be reinitialized. I haven't been able to find a way to do this in the model's config though.

**Comment by AbinayaM02** [comment]
You can save the first fine-tuned model without head. Not sure if this functionality is available directly in simpletransformers library. I have tried something similar using Transformer lib (usinf from_pretrained and save_pretrained methods) directly. And then load the model without head and add new head for the dataset B by setting no. of labels in the model class in simpletransformers. Now, you should be able to fine-tune model for dataset B.

Btw, 6 million and 6000 labels seems really high! I think this is the no. of data points and not labels I guess.

**Comment by ThilinaRajapakse** [comment]
Does the model being loaded in the second step have a name with "checkpoint" in it? If so, renaming it to remove "checkpoint" should fix the issue. (ST uses the model name to figure out if training is being resumed)
